wgpu-types/src/lib.rs

```diff
@@ -681,6 +681,13 @@ bitflags::bitflags! {
         ///
         /// This is a native-only feature.
         const SHADER_INT16 = 1 << 42;
+        /// Allows shaders to use the `early_depth_test` attribute.
+        ///
+        /// Supported platforms:
+        /// - GLES 3.1+
```

Comment:
This support table should be significantly wider

Reply:
And vk/d3d11 need this feature added

Reply:
Indeed, it's just that we only have it implemented for the gles backend in naga.

Reply:
Wait how does the water example work then?

Reply:
No idea 😅

Reply:
As this is going to make the water example not run anymore on most platforms, let's remove the early depth test attribute from it - as it was likely ignored anyway